### ISSUE TITLE
When retrieving OUT params consider the index relative to the sum of all IN and OUT params

### ISSUE THREAD

**Issue by Manuri** [impossibl/pgjdbc-ng#343]
At [1] there is the following method.

```java
  Object get(int parameterIdx) throws SQLException {

    if (parameterIdx < 1 || parameterIdx > outParameterValues.size()) {
      throw PARAMETER_INDEX_OUT_OF_BOUNDS;
    }

    if (command == null) {
      throw new SQLException("statement not executed");
    }

    parameterIdx--;

    Object val = outParameterValues.get(parameterIdx);

    nullFlag = val == null;

    return val;
  }
```
In this method, it is verified that the parameter index is smaller than the count of **all OUT params**. But when registering OUT parameters we register them considering their index relative to all IN/OUT params. 
eg: Say there are 4 params IN, IN, OUT and  INOUT. They will be registered with indices 1,2,3 and 4 respectively. When retrieving the values after the execution, their retrieval indices also should be 1,2,3 and 4. 
But according to the above provided code, when we try to retrieve the 3rd and 4th params, an exception will be thrown saying parameter index is out of bounds.

[1] https://github.com/impossibl/pgjdbc-ng/blob/develop/src/main/java/com/impossibl/postgres/jdbc/PGCallableStatement.java#L285

**Comment by kdubb** [comment]
@Manuri Can you provide a test case for this? I'm hopeful it's fixed in #349  but I'd like to add a test and the description is fairly complex.

**Comment by kdubb** [comment]
This method & referenced check no longer exist as shown. I believe it's fixed and now covered in our normal `CallabaleTest` tests; if you have different results feel free comment and we can re-open.
